### DISCUSSION TITLE
New note type

### DISCUSSION THREAD

**Discussion by jobidon** [TriliumNext/Trilium#3647] [Q&A] [answered]
Is there a way to create a new node type?
I'd like to create a new type of note that would have properties, scripts custom stuff, etc. I am trying to assess if this is something that can easily be added to trillium (js, css, etc) or if it requires modification of the initial source.
Is there a guide to demonstrate how to create new types and demo of how to include them in a running environment?
thanks

**Comment by XXXJetfireXXX** [comment]
If i'm getting this right, there already exists such type of notes: code notes. Specifically CSS, JS_Frontend and JS_Backend can be used to add scripts/custom themes/etc to Trilium app.

**Reply by XXXJetfireXXX** [reply]
I'm not very familiar with JS, but I like to experiment with CSS pretty often, so here's the example of modifying it: https://github.com/zadam/trilium/discussions/3622

**Comment by jobidon** [comment, marked as answer]
thanks for your reply, however that's not quite what I meant. What I'd like to create is a new type that I could call 'Project', and that would list child notes in another sequence (using js and css) as well as calculate various stats about the child notes. So the content is  NOT js/css, but the new type would use those to present the information in a different way.
The 'render' note type might be able to do this, not sure. But there might be more flexibility by creating a new note type.

**Reply by zadam** [reply]
Hi, "render note" should provide you with all the flexibility you need (based on what you describe). 

Combining that with a [template](https://github.com/zadam/trilium/wiki/Template), it sort of creates a new "note type", e.g. allowing creating new instances of this "type" from menu, assigning it some default attributes etc.

![image](https://user-images.githubusercontent.com/617641/221697691-0b965571-2f8f-4c91-876d-c4369549adcf.png)

**Comment by jobidon** [comment]
Indeed, looks like that combination would work for what I want to do: I'll look into that some more. Thank for your insight into this!
